src/test/test_failure_slice.cpp

```
@@ -21,15 +21,16 @@ static std::vector<FailureSlice> get_failure_slices(const std::string& filename,
     REQUIRE(raw_progs.size() == 1);
 
-    const RawProgram& raw_prog = raw_progs.back();
+    RawProgram raw_prog = raw_progs.back();
     auto prog_or_error = unmarshal(raw_prog, options);
     auto inst_seq = std::get_if<InstructionSeq>(&prog_or_error);
     REQUIRE(inst_seq != nullptr);
 
     const Program prog = Program::from_sequence(*inst_seq, raw_prog.info, options);
     auto result = analyze(prog);
+    const AnalysisContext context{prog.info(), options, *prog.info().platform};
 
```

Cppcheck (2.20.0)

[style] The function 'make_increment_counter' is never used.

(unusedFunction)